packages/peregrine/lib/talons/MyAccount/useNewsletterSubscription.js

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { useToasts } from '@magento/peregrine';
+import { useUserContext } from '../../context/user';
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+
+export const useNewsletterSubscription = props => {
+    const { subscribeMutation, customerQuery } = props;
+    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();
+
+    const [subscribe, setSubscribe] = useState(
+        currentUser && currentUser.hasOwnProperty('is_subscribed')
+            ? currentUser.is_subscribed
+            : false
+    );
+    const [isSubmitting, setIsSubmitting] = useState(false);
```

Comment:
Neither of these need to be state, they can be derived from state you already have access to. Once you move to `useQuery` it should look like this.

```js
const {data: subscriptionData} = useQuery(getCustomerSubscriptions);
const isSubscribed = !!data && data.customer.is_subscribed;
```